src/builtin/pre_commit_hooks/check_added_large_files.rs

```diff
@@ -0,0 +1,74 @@
+use crate::git::{intent_to_add_files, is_lfs_file};
+use crate::hook::Hook;
+use crate::run::CONCURRENCY;
+use clap::Parser;
+use futures::StreamExt;
+use std::collections::{HashMap, HashSet};
+
+enum FileFilter {
+    NoFilter,
+    Files(HashSet<String>),
+}
+
+impl FileFilter {
+    fn contains(&self, path: &str) -> bool {
+        match self {
+            FileFilter::NoFilter => true,
+            FileFilter::Files(files) => files.contains(path),
+        }
+    }
+}
+
+#[derive(Parser)]
+struct Args {
+    #[arg(long)]
+    enforce_all: bool,
+    #[arg(default_value = "500")]
+    max_kb: u64,
+}
+
+pub(crate) async fn check_added_large_files(
+    hook: &Hook,
+    filenames: &[&String],
+    _env_vars: &HashMap<&'static str, String>,
+) -> anyhow::Result<(i32, Vec<u8>)> {
+    let entry = shlex::split(&hook.entry).ok_or(anyhow::anyhow!("Failed to parse entry"))?;
+    let args = Args::try_parse_from(entry.iter().chain(&hook.args))?;
+
+    let filter = if args.enforce_all {
+        FileFilter::NoFilter
+    } else {
+        let add_files: HashSet<_> = intent_to_add_files().await?.into_iter().collect();
+        FileFilter::Files(add_files)
+    };
+
+    let mut tasks = futures::stream::iter(filenames.iter().filter(|f| filter.contains(f)))
+        .filter(|&&f| async move { is_lfs_file(f).await.unwrap_or_default() })
```

Comment:
For each file, we spawn a `git` subprocess to check LFS, it might not be very efficient.

Reply:
you mean check all at once?

Reply:
Yes, like `pre-commit-hooks` did:

https://github.com/pre-commit/pre-commit-hooks/blob/31903eabdb97f5a3375fb1ea419ff782de64d7e7/pre_commit_hooks/check_added_large_files.py#L13-L30

Reply:
I think it will combine filter and map together without any efficiency issues, but maybe I was wrong.

Reply:
done